src/main/java/htsjdk/samtools/read/Read.java

```diff
@@ -0,0 +1,1438 @@
+     *
+     * @return the bitwise flag as an integer.
+     */
+    int getFlag();
+
+    /**
+     * Gets the {@link SAMFlag} that are set.
+     *
+     * <p>Equivalent to <b>FLAG</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Default implementation calls {@code SAMFlag.getFlags(getFlag())}.
+     *
+     * @return a set of the flags that are set for the read.
+     */
+    default Set<SAMFlag> getSetFlags() {
+        return SAMFlag.getFlags(getFlag());
+    }
+
+    /**
+     * Checks if the flag is set.
+     *
+     * <p>Default implementation calls {@code flag.isSet(getFlag())}.
+     *
+     * @param flag the flag to test.
+     * @return {@code true} if the flag is set; {@code false} otherwise.
+     *
+     * @throws IllegalArgumentException if the flag is {@code null}.
+     */
+    default boolean isSet(final SAMFlag flag) {
+        return flag.isSet(getFlag());
+    }
+
+    /**
+     * Checks if the flag is unset.
+     *
+     * <p>Default implementation calls {@code flag.isUnset(getFlag())}.
+     *
+     * @param flag the flag to test.
+     * @return {@code true} if the flag is unset; {@code false} otherwise.
+     *
+     * @throws IllegalArgumentException if the flag is {@code null}.
+     */
+    default boolean isUnset(final SAMFlag flag) {
+        return flag.isUnset(getFlag());
+    }
+
+    /**
+     * Sets the bitwise flag.
+     *
+     * <p>Equivalent to <b>FLAG</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Note: it is preferable to use the methods for set/unset specific flags are set (using {@link #setFlag(SAMFlag)} or {@link #unsetFlag(SAMFlag)}),
+     * clear the flags using {@link #clearFlag()}, set/unset the specific flags (using {@link #setFlags(Set) and {@link #unsetFlags(Set)}},
+     * or use the shortcuts.
+     *
+     * @param flag bitwise flag to assign to the read.
+     */
+    void setFlag(final int flag);
+
+    /**
+     * Clears the bitwise flag, un-setting all the bits.
+     *
+     * <p>Equivalent to <b>FLAG</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Default implementation calls {@code setFlag(0)}.
+     */
+    default void clearFlag() {
+        setFlag(0);
+    }
+
+    /**
+     * Adds the flag, either setting it or unsetting it.
+     *
+     * <p>Equivalent to <b>FLAG</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Deafault implementation modifies the {@link SAMFlag} bit of the {@link #getFlag()}, setting it afterwards with {@link #setFlag(int)} .
+     *
+     * @param flag flag to be set or unset.
+     * @param set {@code true} if the flag should be set; {@code false} otherwise.
+     *
+     * @throws IllegalArgumentException if the flag is {@code null}.
+     */
+    default void addFlag(final SAMFlag flag, final boolean set) {
```

Comment:
I think that this should still be called "setFlag", it can be set to True or set to False.

Reply:
I used this nomenclature to do not confound with the `setFlag(SAMFlag)` and `unsetFlag(SAMFlag)` for single flags. Does this make sense?

Reply:
no..add means something very different....